### PR TITLE
Log file sharing

### PR DESCRIPTION
**Pull request by wmaciel**
Adds tool bar to the map activity where you can choose to send your log file to other apps.

### CODE REVIEW

**Comment by wmaciel** [comment]
I took me an awful amount of time to get this working.
Let me know if I am breaking any best practices guideline. I am still new at app development

**Comment by wmaciel** [comment]
It looks like it is complaining about the isClosed attribut of OWDevice.
I wonder why this hasn't failed before, because it has nothing to do with my change, I think.
I'll update the pull request to change the variable type to boolean instead of Boolean. This might solve it.

**Comment by wmaciel** [comment]
Turns out I can change ObservableField<Boolean> to ObservableField<boolean> so I looked further into the problem.

According to [this stackoverflow thread](https://stackoverflow.com/questions/23317876/what-is-the-appropriate-way-to-handle-warning-the-expression-of-type-x-is-boxe) the appropriate way to deal with this warning is to mute it.

While I wait for you guys's feedback on this, I'll look for other ways fix it.

**Comment by kwatkins** [comment]
Yeah, data binding can be a pain, I'll grab and test this.

**Comment by wmaciel** [comment]
@kwatkins , I think my latest commit passed the checks, but if you look into the CI output, there still is a warning.

I tried looking where it is coming from but no luck.
Can you spot it?

**Comment by kwatkins** [comment]
Yeah, I see it but not worried about it, one of those nag warnings that you hope will fix itself at some point :-) - the bool OW checks seem to work fine. I ran the build with your change and it looks good! About to take a quick ride and test the changes w/ master.

**Comment by kwatkins** [comment]
This looks good to me. Also @ebabel the cleanup you did seemed to do us a lot of good, the grc exception is no longer being thrown and BLE connections at least seem to be more stable, not sure why, but it does. 

I did a bit of road warrior testing and all seems to be good. OH I did hit an unSafe(pothole) & nosedive, spilling a box of chicken mcnuggets and spicy pumpkin latte over the road (don't judge :P)

@ebabel passing to you to test/close!

File: app/src/main/java/net/kwatts/powtools/MapActivity.java [hunk 1]
```diff
@@ -69,6 +81,38 @@ protected void onCreate(Bundle savedInstanceState) {
...
+
+        // store action provider
+        MenuItem item = menu.findItem(R.id.menu_item_share);
+        mShareActionProvider = (ShareActionProvider) MenuItemCompat.getActionProvider(item);
+
+        // Create share intent
+        Intent sendIntent = new Intent();
+        sendIntent.setAction(Intent.ACTION_SEND);
+        File logFile = new File(PlainTextFileLogger.getLoggingPath() + "/" + mFileName);
+        Uri uri = FileProvider.getUriForFile(this, "net.kwatts.powtools.fileprovider", logFile);
```
Comment (ebabel) [inline app/src/main/java/net/kwatts/powtools/MapActivity.java]: @wmaciel fyi also could have used Uri.fromFile

**Comment by ebabel** [comment]
Well done @wmaciel. I've never seen a ShareActionProvider before. Looks like more work than what I would have done: https://stackoverflow.com/a/9975439/247325 but probably better in the long run!

**Comment by ebabel** [comment]
@kwatkins sorry about your nuggets :( hope you didn't get hurt!

**Pull request merged**